sklearn/utils/extmath.py

```diff
@@ -273,6 +273,7 @@ def randomized_range_finder(
            [-0.52...,  0.24...],
            [-0.82..., -0.38...]])
     """
+    A = check_array(A, accept_sparse=True)
     xp, is_array_api_compliant = get_namespace(A)
```

Comment:
This is going to basically cancel the array API work I think.

I wonder if explicitly checking for complex values is the only way.

cc @OmarManzoor maybe

Reply:
Ah, you're right. Thanks for the catch. What if we just call `check_array` but not reassign `A`?

Reply:
I think if we don't assign the new array to A, it should work like before while handling the exceptions we need to catch. By the way I think check_array supports the array api itself.

Reply:
``check_array`` is array-api compliant so it can be used as usual.

Reply:
@jeremiedbb, I see, when I tried out `check_array` on a torch Tensor previously, I didn't have the array api enabled.